src/ai/defaultai.cc

```diff
@@ -2823,31 +3014,32 @@ bool DefaultAI::construct_building(const Time& gametime) {
 						prio += 5 * std::abs(management_data.get_military_number_at(17));
 					}
 
-					if (bf->trees_nearby < trees_nearby_treshold_ &&
+					if (bf->immovables_by_name_nearby[bo.name] < trees_nearby_treshold_ &&
 					    bo.new_building == BuildingNecessity::kAllowed) {
 						continue;
 					}
 
 					prio += std::abs(management_data.get_military_number_at(26)) *
-					        (bf->trees_nearby - trees_nearby_treshold_) / 10;
+					        (bf->immovables_by_name_nearby[bo.name] - trees_nearby_treshold_) / 10;
 
 					// consider cutters and rangers nearby
-					prio += 2 * bf->supporters_nearby.at(bo.get_collected_map_resource()) *
+					prio += 2 * number_of_supporters_nearby *
 					        std::abs(management_data.get_military_number_at(25));
-					prio -= bf->collecting_producers_nearby.at(bo.get_collected_map_resource()) *
+					prio -= number_of_supported_producers_nearby *
```

Comment:
as far as I understand the code above this is calculated from the current BuildingObserver which means it is zero for lumberjacks and not identifiying other lumberjacks as it should